plugin/pkg/global/resourcereferencemanager/admission_test.go

```diff
@@ -600,6 +603,84 @@ var _ = Describe("resourcereferencemanager", func() {
 					},
 				})))
 			})
+
+			It("should allow specifying a namespace which is not in use (create)", func() {
+				project.Spec.Namespace = pointer.StringPtr("garden-foo")
+				projectCopy := project.DeepCopy()
+				projectCopy.Name = "project-2"
+				projectCopy.Spec.Namespace = pointer.StringPtr("garden-bar")
+				gardenCoreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&projectCopy)
+
+				attrs := admission.NewAttributesRecord(&project, nil, core.Kind("Project").WithVersion("version"), project.Namespace, project.Name, core.Resource("projects").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, defaultUserInfo)
+
+				err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+				Expect(err).To(Not(HaveOccurred()))
+			})
+
+			It("should allow specifying a namespace which is not in use (update)", func() {
+				projectOld := project.DeepCopy()
+				projectCopy := project.DeepCopy()
+				project.Spec.Namespace = pointer.StringPtr("garden-foo")
+				projectCopy.Name = "project-2"
+				projectCopy.Spec.Namespace = pointer.StringPtr("garden-bar")
+				gardenCoreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&projectCopy)
```

Comment:
On update the informer will already have the old project, right?
So, I guess, it makes sense to add it also in the test, because you are listing all projects in `ensureProjectNamespace ()`.
```suggestion
				gardenCoreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&projectOld)
				gardenCoreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&projectCopy)
```